### PR TITLE
New package: vault-0.10.4

### PR DESCRIPTION
**Pull request by iaroki**

### CODE REVIEW

**Comment by the-maldridge** [comment]
I'd like to see service scripts unless you've got a good reason not to.

File: srcpkgs/vault/files/vault.hcl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+storage "file" {
+	path = "/var/lib/vault"
+}
+
+listener "tcp" {
+	address = "127.0.0.1:8200"
+	tls_disable = 1
+}
+
```
Comment (maxice8) [inline srcpkgs/vault/files/vault.hcl]: extra blank line

File: srcpkgs/vault/template [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+revision=1
+build_style=go
+go_import_path="github.com/hashicorp/${pkgname}"
+short_desc="Tool for securely accessing secrets"
+maintainer="iaroki <iaroki@protonmail.com>"
+license="MPL-2.0"
+homepage="https://www.vaultproject.io/"
+distfiles="https://github.com/hashicorp/${pkgname}/archive/v${version}.tar.gz"
+checksum=5f3e4dd0cd1f4cb1e4fc4be38a4839d8683bad62ad5b7a68f1e49f545750b42b
+
```
Comment (maxice8) [inline srcpkgs/vault/template]: conf_files="/etc/vault.hcl" ?

File: srcpkgs/vault/files/vault.hcl (hunk 1 above)
Comment (the-maldridge) [reply srcpkgs/vault/files/vault.hcl]: This also needs to now be a conf_file

File: srcpkgs/vault/files/vault/run [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+exec vault server -config=/etc/vault.hcl
```
Comment (the-maldridge) [inline srcpkgs/vault/files/vault/run]: I'd rather not have this running as root.  Please create a system user and run it as that instead.

File: srcpkgs/vault/files/vault/run [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec chpst -u vault vault server -config=/etc/vault.hcl
```
Comment (maxice8) [inline srcpkgs/vault/files/vault/run]: problably _vault is better suited like it is done elsewhere

**Comment by the-maldridge** [comment]
Overall looks good, but please don't run as root.  Create a system user '_vault' and run as that.

File: srcpkgs/vault/template [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+revision=1
+build_style=go
+go_import_path="github.com/hashicorp/${pkgname}"
+short_desc="Tool for securely accessing secrets"
+maintainer="iaroki <iaroki@protonmail.com>"
+license="MPL-2.0"
+homepage="https://www.vaultproject.io/"
+distfiles="https://github.com/hashicorp/${pkgname}/archive/v${version}.tar.gz"
+checksum=5f3e4dd0cd1f4cb1e4fc4be38a4839d8683bad62ad5b7a68f1e49f545750b42b
+system_accounts="_vault"
```
Comment (iaroki) [inline srcpkgs/vault/template]: But I already created system user here.

File: srcpkgs/vault/files/vault/run [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec chpst -u _vault vault server -config=/etc/vault.hcl
```
Comment (iaroki) [inline srcpkgs/vault/files/vault/run]: And run as `_vault` user here.
Am I missing something?

Comment (maxice8) [reply srcpkgs/vault/files/vault/run]: shouldn't it have a user and group _vault ?

chpst -u _vault:_vault

Comment (the-maldridge) [reply srcpkgs/vault/files/vault/run]: Leaving the group unset should be fine.  Running as nogroup would be acceptable if the umask is set such that files are only written 0600.

Comment (jnbr) [reply srcpkgs/vault/files/vault/run]: `chpst -u _vault` is fine. It will default to group `_vault`

Comment (the-maldridge) [reply srcpkgs/vault/files/vault/run]: sgtm.

**Pull request merged**